Library/Homebrew/extend/os/mac/hardware/cpu.rb

```diff
@@ -107,6 +115,12 @@ def sse4_2?
         sysctl_bool("hw.optional.sse4_2")
       end
 
+      # Note: this is more reliable than checking uname.
+      # `sysctl` returns the right answer even when running in Rosetta.
+      def arm64?
```

Comment:
```suggestion
      def native_arm64?
```
Alternatively to my previous suggestion, rename this function to `native_arm64?`. `arm64?` should continue to return `false` when running under Rosetta.